packages/itwin/breakdown-trees/src/Views/visibility/DataLink.ts

```diff
@@ -10,100 +10,79 @@ export class DataLink {
-    from spatialcomposition.CompositeOverlapsSpatialElements ovr join biscore.spatialIndex spel on spel.ECinstanceId = ovr.targetecinstanceid
-    where sourceECInstanceId in
-    (select sp.ecinstanceid from buildingspatial.story s join buildingspatial.space sp on s.ecinstanceid=sp.composingelement.id where s.ecinstanceid=${storyId}
-    union
-    select s.ecinstanceid from buildingspatial.story s where s.ecinstanceid=${storyId})`;
-
-    let queryGetStoreyProps;
-    if (clipAtSpaces) {
-      queryGetStoreyProps = `select MIN(spel.minX) as minX, MIN(spel.minY) as minY, MIN(spel.minZ) as minZ, MAX(spel.maxX) as maxX, MAX(spel.maxY) as maxY, MAX(spel.maxZ) as maxZ
-      from spatialcomposition.CompositeComposesSubComposites ovr join biscore.spatialIndex spel on spel.ECinstanceId = ovr.targetecinstanceid
-      where sourceECInstanceId in
-      (select sp.ecinstanceid from buildingspatial.story s join buildingspatial.space sp on s.ecinstanceid=sp.composingelement.id where s.ecinstanceid=${storyId}
-      union
-      select s.ecinstanceid from buildingspatial.story s where s.ecinstanceid=${storyId})`;
-      let rows = await this.executeQuery(iModel, queryGetStoreyProps);
-      if (rows.length > 0 && rows[0].minX)
-        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-      rows = await this.executeQuery(iModel, queryGetStoryWithElementsProps);
-      if (rows.length > 0 && rows[0].minX)
-        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-    } else {
-      queryGetStoreyProps = queryGetStoryWithElementsProps;
-      const rows = await this.executeQuery(iModel, queryGetStoreyProps);
+    const runQuery = async (query: string) => {
+      const rows = await this.executeQuery(iModel, query);
+
       if (rows.length > 0 && rows[0].minX)
         return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-    }
-    return undefined;
+      return undefined;
+    };
+
+    const constructQuery = (queryClass: string) => {
+      return `select MIN(spel.minX) as minX, MIN(spel.minY) as minY, MIN(spel.minZ) as minZ, MAX(spel.maxX) as maxX, MAX(spel.maxY) as maxY, MAX(spel.maxZ) as maxZ
+        from ${queryClass} ovr join biscore.spatialIndex spel on spel.ECinstanceId = ovr.targetecinstanceid
+        where sourceECInstanceId in
+        (select sp.ecinstanceid from buildingspatial.story s join buildingspatial.space sp on s.ecinstanceid=sp.composingelement.id where s.ecinstanceid=${storyId}
+        union
+        select s.ecinstanceid from buildingspatial.story s where s.ecinstanceid=${storyId})`;
+    };
+
+    return clipAtSpaces && runQuery(constructQuery("spatialcomposition.CompositeComposesSubComposites")) || runQuery(constructQuery("spatialcomposition.CompositeOverlapsSpatialElements"));
```

Comment:
this may be more compact, but has poor readability

Reply:
Yes I think we talked about this as team and concluded that we'll try to DRY as much as possible. @JuliusArmalis any thoughts?

Reply:
DRY goes for "Don't repeat yourself" not squeeze everything into a single line.

i'd make it 2 lines:

const query = clipAtSpaces ? constructQuery("spatialcomposition.CompositeComposesSubComposites") : constructQuery("spatialcomposition.CompositeOverlapsSpatialElements");

return runQuery(query);

//or something similar

Reply:
This wouldn't work since if clipAtSpaces is true then we need to run the query with CompositeComposesSubComposites but if that query returns undefined then we need to run the query again with CompositeOverlapsSpatialElements. I could change it as following

const compositeComposesQuery = constructQuery("spatialcomposition.CompositeComposesSubComposites");
const compositeOverlapsQuery = constructQuery("spatialcomposition.CompositeOverlapsSpatialElements");

return clipAtSpaces && runQuery(compositeComposesQuery) || runQuery(compositeOverlapsQuery);

Reply:
I think that using named variables to increase readability is a good option here.

Reply:
i think that statement "if that query returns undefined then we need to run the query again with CompositeOverlapsSpatialElements" is why this needs to be organized better. For me I missed that detail as I was reviewing, and good code should not need documentation.

I think something like this would be more readable? :

let result = undefined;

if (clipAtSpaces)
	{
	result = runQuery(constructQuery("spatialcomposition.CompositeComposesSubComposites"));
	}

return result ?? runQuery(constructQuery("spatialcomposition.CompositeOverlapsSpatialElements"));

Reply:
I would totally agree on this one because we should prioritize readability

Reply:
Fixed